[draft] Proposal: Sub-Resources for CustomResources

nikhita opened the pull request.

verb committed:
Propose a feature to troubleshoot running pods

This feature allows troubleshooting of running pods by running a new
"Debug Container" in the pod namespaces.

verb committed:
Address feedback in pod troubleshooting proposal

The biggest change is a reworking of the Kubernetes API changes after
chatting with sig-api-machinery. The Debug Container is now created in a
standalone API call and the client is responsible for attaching.

Commit:
Allow specifying the primary group id of the container

verb committed:
Add analysis of standalone pod alternative

kksriram committed:
Vaultproviderspec (#1)

* Initial version of Vault KMS Provider

* Renamed image file to be consistent with spec name

* Brought spec inline with EnvelopeTransformer interfaces in PR 49350

* Wrap at 80

kksriram committed:
resolved conflicts

Commit:
add PodSecurityPolicy changes plus review comments

nikhita committed:
Proposal: SubResources for CustomResources

nikhita commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+### Feature Gate
+
+The `SubResources` field in `CustomResourceDefinitionSpec` will be gated under the `CustomResourceSubResources` alpha feature gate.
+If the gate is not open, the value of the new field within `CustomResourceDefinitionSpec` is dropped on creation and updates of CRDs.
```
dropped => rejected?

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+[CustomResourceDefinitions](https://github.com/kubernetes/community/pull/524) (CRDs) were introduced in 1.7. The objects defined by CRDs are called CustomResources (CRs). Currently, we do not provide sub-resources for CRs.
+
+However, it is one of [most requested features](https://github.com/kubernetes/kubernetes/issues/38113) and this proposal seeks to add  `/status `and `/scale` subresources for CustomResources.
```
we have to use one spelling: sub-resources or subresources. I would choose the later.

sttts replied:
We also have "sub-object" a couple of times. Should also be "subobject."

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+## Non-Goals
+
+1. Allow defining arbitrary subresources i.e. subresources except `/status` and `/scale`.
+2. Unify the many Scale types: as long as there is no generic Scale object in Kubernetes, we will propose to introduce yet another Scale type in the `apiextensions.k8s.io` api group. If Kubernetes switches to a generic Scale object, `apiextensions.k8s.io` will follow.
```
quoted `Scale`

nikhita commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+Only `ScaleSpec.Replicas` can be written. All other values are read-only and changes will be ignored. i.e. upon updating the scale sub-resource, two fields are modified:
+
+1. The replicas field is copied back from the Scale object to the main resource as specified by `SpecReplicasPath` in the CRD, e.g.  `.spec.replicas = scale.Spec.Replicas`.
```
will make it `Scale` here (and everywhere).

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+    // label query over pods that should match the replicas count.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    // +optional
+    Selector *metav1.LabelSelector
```
json tags

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+    metav1.TypeMeta `json:",inline"`
+    // Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata.
+    // +optional
+    metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```
protobuf tags are missing for a lot of fields. I would drop them from the proposal and add them later in code.

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+### Feature Gate
+
+The `SubResources` field in `CustomResourceDefinitionSpec` will be gated under the `CustomResourceSubResources` alpha feature gate.
+If the gate is not open, the value of the new field within `CustomResourceDefinitionSpec` is dropped on creation and updates of CRDs.
```
didn't we opt for rejection?

nikhita replied:
Yes, will update it.

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+#### Status
+
+The status endpoint of a CustomResource receives a full CR object. Changes outside of the `StatusPath` are ignored.
```
would just write `status path`. Otherwise, we have to write "outside of what is specified in `StatusPath`.  Both is fine.

sttts replied:
or as below: value at the specified JSON Path `StatusPath`

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+The status endpoint of a CustomResource receives a full CR object. Changes outside of the `StatusPath` are ignored.
+For validation everything outside of the `StatusPath` can be reset to the existing values of the CustomResource. Then the JSON Schema presented in the CRD is validated against the whole object.
+
+Note: one could extract the status sub-object, filter the JSON schema by rules applying to the status and then validate the status sub-object only.
```
applying -> applied

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+Moreover, if the scale subresource is enabled:
+
+- The value at the specified JSON Path `SpecReplicasPath` (e.g. `.spec.replicas`) is validated to ensure that it contains an integer value and is not empty.
```
non-negative integer value

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+### Scale Behavior
+
+The number of CustomResources can be easily scaled up or down depending on the replicas field present in `.spec`.
```
in path specified by `SpecPath` (e.g. `.spec`).

nikhita commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+    // The JSON path (e.g. “.status”) of the status of a CustomResource.
+    // The whole object is transferred over the wire, but only the status can be mutated via the /status subresource.
+    // StatusPath is restricted to be “.status” and defaults to “.status”.
+    StatusPath string `json:“path,omitempty”`
```
`statusPath` and `specPath`.

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+#### Selector Behavior
+
+`CustomResourceSubResourceScale.LabelSelectorPath` is the label selector over CustomResources that should match the replicas count.
+The value in the Scale object is the one-to-one the value from the CustomResource if the label selector is non-empty.
```
is one-to-one the value

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+`CustomResourceSubResourceScale.LabelSelectorPath` is the label selector over CustomResources that should match the replicas count.
+The value in the Scale object is the one-to-one the value from the CustomResource if the label selector is non-empty.
+Intentionally we do not default it to another value from the CustomResource (e.g. `.spec.template.metadata.labels`) as this turned out to cause trouble (e.g. in kubectl apply) and it is generally seen as a wrong approach with existing resources.
```
quoted `kubectl apply`

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+### Scope
+
+In this proposal we opted for an opinionated concept of subresources i.e. we restrict the subresource spec to the two very specific subresources: `/status` and `/scale`.
+We do not aim for a more generic subresource concept. In Kubernetes there are number of other subresources like `/log`, `/exec`, `/bind`. But their semantics is much more special than `/status` and `/scale`.
```
a number

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+In this proposal we opted for an opinionated concept of subresources i.e. we restrict the subresource spec to the two very specific subresources: `/status` and `/scale`.
+We do not aim for a more generic subresource concept. In Kubernetes there are number of other subresources like `/log`, `/exec`, `/bind`. But their semantics is much more special than `/status` and `/scale`.
+Hence, we decided to leave those subresources to the domain of User provided API Server (UAS) instead of inventing a more complex subresource concept for CustomResourceDefinitions.
```
those other

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+
+We also restrict the JSON path for the status and the spec within the CustomResource.
+We could make them definable by the user and the proposed types actually allow us to open this up in the future.
+For the time being we decided to be opinionated as all status and spec sub-object in existing types live under `.status` and `.spec`. Keeping this pattern imposes consistency on user provided CustomResources as well.
```
sub-objects

sttts commented on contributors/design-proposals/customresources-subresources.md:
```diff
@@ -0,0 +1,221 @@
+    Validation *CustomResourceValidation
+    // SubResources describes the subresources for CustomResources
+    // This field is alpha-level and should only be sent to servers that enable
+    // SubResources via the CurstomResourceSubResources feature gate.
```
maybe "subresources" here

nikhita replied:
:+1:

gmarek committed:
Add Events design proposal

Commit:
Propose adopting the Rust RFC process

I believe it is time to adopt a new development process given the
current scale of Kubernetes. The inadequacy of our current process
is particularly apparent within SIGs with cross project concerns
such as SIG Release, SIG Testing, SIG Architecture, and SIG PM
so additional process around proposing changes to Kubernetes is
suggested.

Commit:
Update links in proposal to adopt RFC process

Adds links to Google Groups discussion and self link to PR

msau42 committed:
1.7 Local Persistent Volume design doc

Commit:
fix I's

Commit:
fix some formatting

Commit:
add some text about primary group id

Commit:
review comments

Commit:
extract KEP template from KEP proposal and address comments

- RFC -> KEP
- attempt to remove first person tone
- add possible states of a KEP

grodrigues3 committed:
first pass at organizing design proposals folder

Commit:
Creates template for a KEP

grodrigues3 committed:
moved but forgot to commit

grodrigues3 committed:
add-a-text-file-with-dir-struct

Commit:
review comments

verb committed:
Switch Pod Troubleshooting API back to /exec

This updates the Pod Troubleshooting Design Proposal for recent
developments in the community and to reflect the consensus from the API
review: using the existing /exec endpoint as a starting point for this
feature.

grodrigues3 committed:
addressing Brians comments

krisnova committed:
Create kris-nova_bio.md

Commit:
add changes needed

grodrigues3 committed:
addressing claytons comments

Commit:
Alex Pollitt bio for steering committee

ScorpioCPH committed:
Minor fix quick link

idvoretskyi committed:
Removed company affiliation

I'm not an employee of Mirantis since the last week. Removed affiliation.

Commit:
Merge pull request #1042 from ScorpioCPH/master

Automatic merge from submit-queue

Minor fix quick link

@RenaudWasTaken @jiayingz PTAL.

zehicle committed:
Rob Hirschfeld BIO for election

Commit:
Merge pull request #1039 from mattfarina/mattfarina-2017-elections

Automatic merge from submit-queue

Adding matt farina bio details for 2017 elections

Commit:
Merge pull request #1045 from zehicle/master

Automatic merge from submit-queue

Rob Hirschfeld BIO for election

Here is my bio for Steering Committee elections.

As per request of @parispittman

Commit:
Merge pull request #1044 from kubernetes/idvoretskyi-patch-1

Automatic merge from submit-queue

Removed company affiliation

timothysc committed:
timothysc bio

Commit:
Merge pull request #1046 from timothysc/timothysc_bio

Automatic merge from submit-queue

timothysc bio

Adds bio to steering election 

/cc @castrojo

jdumars committed:
Added Jaice Singer DuMars bio

jdumars committed:
added bio to README

sjenning committed:
correct Michael Rubin github handle

jdumars committed:
update SIG Architecture meeting day and time

Commit:
Merge pull request #1050 from kubernetes/sjenning-patch-1

Automatic merge from submit-queue

correct Michael Rubin github handle

jdumars committed:
Ran the generator to update sig-architecture

Commit:
Merge pull request #1051 from jdumars/archmtg

Automatic merge from submit-queue

update SIG Architecture meeting day and time

Changed meeting time to Thursdays at 8:30 PST
CC: @bgrant0607

jdumars committed:
less words

jdumars committed:
even less words

Commit:
Merge pull request #1049 from jdumars/jaicesingerdumars

Automatic merge from submit-queue

Added Jaice Singer DuMars bio

Commit:
add bio for Adnan Abdulhussein

Commit:
Adding the working group for the cluster API effort.

Commit:
Merge pull request #1053 from roberthbailey/wg-cluster-api

Automatic merge from submit-queue

Adding the working group for the cluster API effort.

@luxas fyi

msau42 committed:
Volume topology aware scheduling design

Commit:
Add link to Alex Pollitt bio

Commit:
Merge pull request #1040 from lxpollitt/lxpollitt-bio-1

Automatic merge from submit-queue

Alex Pollitt bio for steering committee voter guide

Commit:
Update OWNERS

Removed czahedi; no longer working on the project

luxas committed:
Add @luxas as a SIG Cluster Lifecycle lead

Commit:
review comments

Commit:
Merge pull request #1056 from prydonius/adnanabdulhussein_bio

Automatic merge from submit-queue

add bio for Adnan Abdulhussein

Commit:
Merge pull request #1057 from parispittman/patch-2

Automatic merge from submit-queue

Update OWNERS

sebgoa committed:
add sebgoa

Commit:
Merge pull request #1055 from luxas/luxas_sig_lead

Automatic merge from submit-queue

Add @luxas as a SIG Cluster Lifecycle lead

@roberthbailey @jbeda @lukemarsden

nielsole committed:
Removing prom-push-gateway flag

The flag does not work anymore as per https://github.com/kubernetes/kubernetes/issues/45947

justinsb committed:
Election guide: add justinsb

justinsb committed:
Include the non-electioneering commitment

Rather than saying that the committee does not endorse any
of the candidates, which means something rather different!

Also remove the term "current bootstrap committee" - the future
committee is the steering committee.

justinsb committed:
Election: Amend bio links statement

Don't impose a naming policy on people

justinsb committed:
election: s/Member In Standing/Member Of Standing/g

justinsb committed:
Election: Avoid commitment to _immediately_ publish

Avoid setting up a promise that may be difficult to meet technically.

justinsb committed:
Fix timezone for zoom meeting

Commit:
Merge pull request #1065 from justinsb/member_in_of_standing

Automatic merge from submit-queue

election: s/Member In Standing/Member Of Standing/g

Commit:
Merge pull request #1066 from justinsb/avoid_time_commitment

Automatic merge from submit-queue

Election: Avoid commitment to _immediately_ publish

Commit:
Merge pull request #1067 from justinsb/sigaws_time

Automatic merge from submit-queue

Fix timezone for sig-aws zoom meeting

Commit:
Merge pull request #1059 from sebgoa/sebgoa

Automatic merge from submit-queue

add sebgoa

Adds sebastien goasguen's bio to the ballot table :)

Commit:
Merge pull request #1063 from justinsb/bootstrap_consistency

Automatic merge from submit-queue

Include the non-electioneering commitment

Commit:
Merge pull request #1062 from justinsb/vote_for_justinsb

Automatic merge from submit-queue

Election guide: add justinsb

bgrant0607 committed:
Removed events from backlog

Commit:
clarify behavior

grodrigues3 committed:
addressed pr comments

Commit:
Merge pull request #1061 from nielsole/patch-1

Automatic merge from submit-queue

Removing prom-push-gateway flag

arschles committed:
Adding bio for Aaron Schlesinger

arschles committed:
splitting up language in about the section

arschles committed:
adding link to my bio

Commit:
Added link to SIG-docs youtube channel

Commit:
Added video URL to sigs.yaml

arschles committed:
trimming down the number of words

Commit:
Updated frequency to weekly

arschles committed:
shortening more

Commit:
Merge pull request #1070 from arschles/aaron-bio

Automatic merge from submit-queue

Adding bio for Aaron Schlesinger

grodrigues3 committed:
removing dupe hugepages and obsolete marking

Commit:
Generated output of make

Commit:
Merge pull request #1071 from zacharysarah/1052-link-to-videos

Automatic merge from submit-queue

Added link to SIG-docs youtube channel

Closes #1052 

This PR:
- Updates the link to sig-docs meeting videos with an actual link to videos

bgrant0607 committed:
Merge pull request #1010 from grodrigues3/org-design-propals

Organize Design Proposals Dir by SIG

bgrant0607 committed:
Delete obsolete image

bgrant0607 committed:
Delete obsolete image

bgrant0607 committed:
Fix architecture.md link

spiffxp committed:
Add Aaron Crickenberger bio

Commit:
Add Doug's bio

Signed-off-by: Doug Davis <dug@us.ibm.com>

Commit:
Merge pull request #1072 from spiffxp/vote-for-spiffxp-2017

Automatic merge from submit-queue

Add Aaron Crickenberger bio

Commit:
Merge pull request #1073 from duglin/DugBio

Automatic merge from submit-queue

Add Doug's bio

Commit:
Merge pull request #1064 from justinsb/dont_impose_name_policy

Automatic merge from submit-queue

Election: Amend bio links statement

* Don't impose a naming policy on people

* Candidates were not asked at the same time to provide bios, so avoid
statements that could give the wrong impression.

sakeven committed:
fix architecture link

Signed-off-by: sakeven <jc5930@sina.cn>

Commit:
Merge pull request #1074 from sakeven/fix/architecture_link

Automatic merge from submit-queue

fix architecture link

Signed-off-by: sakeven <jc5930@sina.cn>

The link of architecture is broken. Just fix it.

derekwaynecarr committed:
Derek Carr - Steering Committee Bio

Commit:
Merge pull request #1076 from derekwaynecarr/bio

Automatic merge from submit-queue

Derek Carr - Steering Committee Bio

Commit:
Update README

Added more voter instructions and fixed format on eligibility.

Commit:
Fix broken link for federation/federation proposal

Commit:
Merge pull request #1079 from budhrg/master

Automatic merge from submit-queue

Fix broken link for federation/federation proposal

ScorpioCPH committed:
Minor fix typo

Commit:
Merge pull request #1081 from ScorpioCPH/fix-typo

Automatic merge from submit-queue

Minor fix typo

@RenaudWasTaken @jiayingz PTAL :)

ianchakeres committed:
Fixed links to principles.md

Commit:
Merge pull request #1082 from ianchakeres/fix-link-principles

Automatic merge from submit-queue

Fixed links to architecture.md and principles.md

Looks like principles.md and architecture.md, moved into the subdirectory architecture a few weeks ago. The links no longer work.

This PR fixes the links to these documents.

Commit:
Merge pull request #756 from krmayankk/runasg

Automatic merge from submit-queue

Allow specifying the primary group id of the container

Allows specifying the Primary Group ID of the container, in addition to Primary User Id when running in Docker similar to how docker supports this.

Commit:
Merge pull request #1077 from parispittman/patch-3

Automatic merge from submit-queue

Update README

mwielgus committed:
Move sig-autoscaling vc call to Monday 14:00 UTC

Commit:
Add Michael Rubin bio

Commit:
Merge pull request #1084 from matchstick/master

Automatic merge from submit-queue. .

Add Michael Rubin bio

Adding Michael Rubin bio to 2017 election.

zouyee committed:
fix version.md link

zouyee committed:
update API group  link in resource-metrics-api.md

zouyee committed:
update aggregated-api-servers.md link

Commit:
Merge pull request #1085 from zouyee/version-fix

Automatic merge from submit-queue. .

fix version.md link

Commit:
Merge pull request #1086 from zouyee/resource

Automatic merge from submit-queue. .

update API group  link in resource-metrics-api.md

Commit:
Merge pull request #1087 from zouyee/aggerated-api-servers

Automatic merge from submit-queue. .

update aggregated-api-servers.md link

update aggregated-api-servers.md link which has been moved to api-machinery

zouyee committed:
delete outdated link

Commit:
Merge pull request #1083 from mwielgus/sig-auto-time

Automatic merge from submit-queue. .

Move sig-autoscaling vc call to Monday 14:00 UTC

The change has been approved by all attending sig members.

idvoretskyi committed:
Company affiliation updated

idvoretskyi committed:
Update README.md

idvoretskyi committed:
Company affiliation updated

idvoretskyi committed:
Company affiliation updated

Commit:
Merge pull request #1089 from kubernetes/idvoretskyi-cncf-upd

Automatic merge from submit-queue. .

idvoretskyi company affiliation updated

Having joined CNCF as a [Developer Advocate](https://www.cncf.io/blog/2017/09/18/meet-cncfs-newest-developer-advocate/), updating the affiliation mentions.

pwittrock committed:
Update bio for Phillip Wittrock

Commit:
fix broke link in csi-new-client-library-procedure.md

Commit:
Merge pull request #1090 from pwittrock/voter-guide

Automatic merge from submit-queue. .

Update bio for Phillip Wittrock

justinsb committed:
Update justinsb bio for word limit

idvoretskyi committed:
Add files via upload

idvoretskyi committed:
Ihor Dvoretskyi' bio added

Commit:
Merge pull request #1093 from kubernetes/idvoretskyi-bio-added

Automatic merge from submit-queue. .

idvoretskyi bio added

Commit:
Update README

updated with new information about election email

Commit:
Merge pull request #1094 from parispittman/patch-3

Automatic merge from submit-queue. .

Update README

Commit:
Merge pull request #1092 from justinsb/justinsb_bio_300

Automatic merge from submit-queue. .

Update justinsb bio for word limit

cblecker committed:
Fix sigs.yaml from #1093

Commit:
Merge pull request #1096 from cblecker/ihor-fix

Automatic merge from submit-queue. .

Fix sigs.yaml from #1093

brgnepal committed:
Fix sig-cli meeting recordings url

Commit:
Merge pull request #1095 from budhrg/fix-sig-cli-recording

Automatic merge from submit-queue. .

Fix sig-cli meeting recordings url

Previously some random video was playing on clicking the hyperlink for meeting recordings.

Commit:
add bio for steering committee elections(michelle)

Commit:
Merge pull request #1097 from michelleN/michellenoorali-bio

Automatic merge from submit-queue. .

add bio for steering committee elections(michelle)

Commit:
add link to bio (michelle)

grodrigues3 committed:
keep consistent naming convention for sigdirs

Commit:
Merge pull request #1098 from michelleN/add-link

Automatic merge from submit-queue. .

add link to bio (michelle)

Commit:
Merge pull request #1099 from grodrigues3/making-naming-consistent

Automatic merge from submit-queue. .

keep consistent naming convention for sigdirs

None of the other sigs had a sig- prefix so changing sig-cli to match.

quinton-hoole committed:
Add bio for Quinton Hoole.

quinton-hoole committed:
Add link to Quinton's bio from README.md

Commit:
Merge pull request #1101 from quinton-hoole/2017-09-19-quintonhoole_bio

Automatic merge from submit-queue. .

Add bio for Quinton Hoole.

cblecker committed:
Add link to Ihor's bio

marun committed:
Update active milestone issues process to reflect bot changes

Commit:
Merge pull request #1102 from kubernetes/cblecker-patch-1

Automatic merge from submit-queue. .

Add link to Ihor's bio

Commit:
Update the pointer to a new doc and create a link to archived document

Commit:
Update the yaml instead of the raw files

yuexiao-wang committed:
Clean up the bad link

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

bobsongplus committed:
fix bad link

Commit:
Merge pull request #1107 from TinySong/fix_flexvolume_bad_link

Automatic merge from submit-queue. .

fix bad link

fix flexvolume bad link

Commit:
Merge pull request #1016 from xiangpengzhao/fix-doc

Automatic merge from submit-queue. .

Update Development Automation and Pull Request Process

fixes: #834 

/cc @bgrant0607 @spiffxp

Commit:
Merge pull request #1103 from vaikas-google/master

Automatic merge from submit-queue. .

Update the pointer to a new doc and create a link to archived document

Commit:
Merge pull request #1106 from yuexiao-wang/fix-link

Automatic merge from submit-queue. .

Clean up the bad link

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>

1.  Remove the title link
2.  Modify the links which don't work

bgrant0607 committed:
Prioritized, updated, reorganized backlog

ScorpioCPH committed:
Minor fix format

Commit:
Merge pull request #1108 from ScorpioCPH/minor-fix-format

Automatic merge from submit-queue. .

Minor fix format

@vishh @RenaudWasTaken PTAL :-)

xiangpengzhao committed:
Update merge requirements in doc

ScorpioCPH committed:
Highlight keywords and fix typo

xiangpengzhao committed:
Merge pull request #1111 from ScorpioCPH/highlight-keywords

Highlight keywords and fix typo

xiangpengzhao committed:
Merge pull request #1091 from u2takey/master

fix stale link

Commit:
Rename security-context-constraints.md to pod-security-policy.md

Aso removes Google Analytics link.

Commit:
Move runas-groupid proposal to auth subdir.

Commit:
flex-volumes-drivers-psp.md: move from storage/ to auth/

alwqx committed:
fix typos

Commit:
Fix broken links after moving proposals to subdirs.

Commit:
Merge pull request #1115 from dockerq/master

Automatic merge from submit-queue. .

fix typos

fix typos in `contributors/design-proposals/cluster-lifecycle/local-cluster-ux.md`

Commit:
Merge pull request #1114 from php-coder/move_flex_psp_proposal

Automatic merge from submit-queue. .

flex-volumes-drivers-psp.md: move from storage/ to auth/

Follow-up to #1010

CC @grodrigues3 @simo5

Commit:
Merge pull request #1113 from php-coder/rename_runas_groupid

Automatic merge from submit-queue. .

Move runas-groupid proposal to auth subdir

Follow-up to #1010

CC @grodrigues3 @simo5

bgrant0607 committed:
Merge pull request #1116 from php-coder/fix_broken_links

Fix broken links after moving proposals to subdirs

Commit:
Fix broken links.

Commit:
Merge pull request #1118 from php-coder/fix_new_broken_be_me_links

Automatic merge from submit-queue. .

Fix broken links

Fix broken links after https://github.com/kubernetes/community/pull/1116

PTAL @mindprince 
CC @simo5

Commit:
Merge pull request #1112 from php-coder/rename_scc_to_psp

Automatic merge from submit-queue. .

Rename security-context-constraints.md to pod-security-policy.md

It's counter intuitive that proposal about PSP has name `security-context-constraints.md`.

@pweil- Do you have objections against renaming a file?

CC @simo5

tengqm committed:
Correct meeting time for Resource Management WG

Commit:
Merge pull request #1120 from tengqm/rm-wg-meeting-time

Automatic merge from submit-queue. .

Correct meeting time for Resource Management WG

mattfarina committed:
Adding SIG Apps subgroup meeting details for helm and charts

Commit:
Merge pull request #1123 from mattfarina/sig-apps-subgroups

Automatic merge from submit-queue. .

Adding SIG Apps subgroup meeting details for helm and charts

@prydonius @michelleN Trying to make these meetings just a little more findable.

Commit:
Merge pull request #1110 from xiangpengzhao/update-doc

Automatic merge from submit-queue. .

Update merge requirements in doc

fixes: https://github.com/kubernetes/community/pull/1016#issuecomment-330992673

/cc @spiffxp

Commit:
Merge pull request #1088 from zouyee/link-outdate

Automatic merge from submit-queue. .

delete outdated link

krisnova committed:
Finally updating this thing

krisnova committed:
Trim to 300

Commit:
Merge pull request #1022 from kris-nova/patch-1

Automatic merge from submit-queue. .

Kris Nova bio for steering committee

Update description of my bio for the Kubernetes steering community.

Commit:
Update KEP development process with link to KEP template

verb committed:
Move Pod Troubleshooting proposal into node subdir

mattfarina committed:
Restore sig apps zoom url

Commit:
Merge pull request #1125 from mattfarina/restore-sig-apps-meeting-url

Automatic merge from submit-queue. .

Restore sig apps zoom url

When the README and other files for a SIG were autogenerated the zoom meeting URL for sig apps changed. This restores to the previous URL.

cblecker committed:
Update generator to use WHAT variable

Commit:
address more comments

Commit:
Merge pull request #861 from bsteciuk/sig-windows

Automatic merge from submit-queue. .

Update sig-windows to re-add links that were lost in format change.

calebamiles committed:
moves KEP template to architecture design proposals

calebamiles committed:
address comments

from @bgrant0607:
- https://github.com/kubernetes/community/pull/1124#discussion_r140889337
- https://github.com/kubernetes/community/pull/1124#discussion_r140894763
- https://github.com/kubernetes/community/pull/1124#discussion_r140897547

calebamiles committed:
move KEP process proposal to architecture directory.

...also change filename to match title

calebamiles committed:
address comments

from: @bgrant0607
- https://github.com/kubernetes/community/pull/967#issuecomment-332007969

gaocegege committed:
podaffinity.md: Fix a broken link

Signed-off-by: Ce Gao <ce.gao@outlook.com>

mattfarina committed:
Fixed spelling error

Commit:
Merge pull request #1131 from mattfarina/spelling-nit

Automatic merge from submit-queue. .

Fixed spelling error

Just nit picking a spelling mistake

bgrant0607 committed:
Merge pull request #967 from calebamiles/propose-adopting-rust-rfc-process

Propose adopting the Rust RFC process

Commit:
Merge pull request #1124 from calebamiles/propose-kep-template

Automatic merge from submit-queue. .

Propose KEP template

Propose a template for the [KEP process][]


[KEP process]: https://github.com/kubernetes/community/pull/967

Commit:
Merge pull request #1130 from gaocegege/fix-link

Automatic merge from submit-queue. .

podaffinity.md: Fix a broken link

jbeda committed:
Clarify relationship of KEP to release note process

Commit:
Merge pull request #1133 from jbeda/kep-focus

Automatic merge from submit-queue. .

Clarify relationship of KEP to release note process

I'm opening this up to have a discussion on what the relationship between a KEP and the release note process is.  I see KEPs being most useful for things that will often span releases.  I *want* us to think bigger and across multiple releases.  Often times a KEP will be discussed and accepted before and outside of the release cycle.

I've also moved some stuff around so that this doc stands more on its own vs. being a "delta" on our current process.  In a years time this will be our current process and this won't read as clearly.

Commit:
Merge pull request #1127 from cblecker/generator-what

Automatic merge from submit-queue. .

Update generator to use WHAT variable

Having to set two different variables is confusing. Changed this to look at the `WHAT` envvar and match on a suffix.

Fixes #1126.

cc: @jamiehannaford

Commit:
Merge pull request #932 from gmarek/events

Automatic merge from submit-queue.

Add Events design proposal

verb committed:
Name PodExecOptions fields as recommended by API reviewer

jbeda committed:
KEP: markdownify metadata items

jbeda committed:
Rename KEP process file to match the defined process.

Commit:
Merge pull request #649 from verb/pod-troubleshooting

Automatic merge from submit-queue.

Propose a feature to troubleshoot running pods

This feature allows troubleshooting of running pods by running a new "Debug Container" in the pod namespaces.

This proposal was originally opened and reviewed in kubernetes/kubernetes#35584.

This proposal needs LGTM by the following SIGs:
- [ ] SIG Node
- [ ] SIG CLI
- [ ] SIG Auth
- [x] API Reviewer

Work in Progress:
- [x] Prototype `kubectl attach` for debug containers
- [x] Talk to sig-api-machinery about `/debug` subresource semantics

Commit:
Merge pull request #929 from tcharding/kubectl-conventions

Automatic merge from submit-queue.

Update Example conventions

**Why this PR is needed**
Currently `kubectl-conventions.md` contains incorrect conventions for command examples. File currently stipulates that example commands should start with a `$`, this is incorrect.

Currently conventions stipulate that command example comments should terminate with a period. This is overly restrictive and trivial. Conventions would be better if this was relaxed to allow developer discretion.

**What this PR does**
Remove `$` convention and stipulate explicitly that command examples **do not** start with a `$`. Remove convention stipulating command examples terminate with a period.

cblecker committed:
fix link in how-to-doc.md

saad-ali committed:
Merge pull request #989 from msau42/local-volume-design

1.7 Local Persistent Volume design doc

Commit:
Merge pull request #1139 from cblecker/link-fix-1

Automatic merge from submit-queue.

fix link in how-to-doc.md

fixes #1080.

Commit:
Merge pull request #1012 from thockin/master

Automatic merge from submit-queue.

Rewrite docs on godep

This corresponds to https://github.com/kubernetes/kubernetes/pull/51766 - do not merge until we get confirmation on that PR.

jbeda committed:
Put instructions at start of KEP template

jbeda committed:
Merge instructions into KEP template.

gmarek committed:
Move Events redesign proposal to instrumentation design dir

Commit:
Merge pull request #1141 from gmarek/events

Automatic merge from submit-queue.

Move Events redesign proposal to instrumentation design dir

Commit:
Merge pull request #1135 from jbeda/kep-metadata

Automatic merge from submit-queue.

KEP metadata

Markdownify KEP metadata proposal that was discussed in https://docs.google.com/document/d/1ynmBMuDuT7yGzRscObB1KtgJj8ljYq0I5q4oshrJUCs/edit?ts=59c1b868#.

Also moved files around so that we are following some of the suggestions here.  Merged the KEP template and the template instructions to reduce the number of files we need to manage.

pwittrock committed:
Proposal for re-factoring apply

- define interfaces for encapuslating logic single-purpose in structs
- use PUT instead of PATCH so we don't need to generate PATCH directives

msau42 committed:
Add downsides of proposed approach, add more details about alternative approach and its downsides, add more details about binding rollback using PV taints

rohitagarwal003 committed:
Don't retire kube-state-metrics. Retire md-check.

kube-state-metrics is active once again (it became active soon after the first commit to this file when it was proposed to be retired). It seems to be the only popular tool that does what it does.

Decision to retire md-check was taken in https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubernetes-dev/0FcCGNtnxE0/JmQfETD0AwAJ

Commit:
Merge pull request #1143 from mindprince/patch-1

Automatic merge from submit-queue.

Don't retire kube-state-metrics. Retire md-check.

kube-state-metrics is active once again (it became active soon after the first commit to this file when it was proposed to be retired). It seems to be the only popular tool that does what it does.

Decision to retire md-check was taken in https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubernetes-dev/0FcCGNtnxE0/JmQfETD0AwAJ

Commit:
Merge pull request #1028 from pwittrock/apply2

Automatic merge from submit-queue.

Proposal for re-architecting apply

Commit:
Merge pull request #1104 from marun/update-issues-process

Automatic merge from submit-queue.

Update active milestone issues process to reflect bot changes

The bot changes are still in review:  https://github.com/kubernetes/test-infra/pull/4433

Commit:
Merge pull request #1054 from msau42/storage-topology-design

Automatic merge from submit-queue.

Volume topology aware scheduling design

Proposal for a smarter scheduler that influences PV binding.

Part of kubernetes/features#121

/sig storage
/sig scheduling
/cc @kubernetes/sig-storage-proposals @kubernetes/sig-scheduling-proposals

marun committed:
Update active milestone issues process to reflect new labels

countspongebob committed:
Update README.md

countspongebob committed:
Merge pull request #1 from countspongebob/countspongebob-patch-1

Update README.md - shift to bi-weekly

Commit:
Merge pull request #1149 from countspongebob/master

Automatic merge from submit-queue.

Updating Scalability page to reflect shift to biweekly schedule.

Updating Scalability page to reflect shift to biweekly schedule.

kksriram committed:
Typos addressing @ericchiang comments

cblecker committed:
Fix sig-scalability doc

Commit:
Merge pull request #1150 from cblecker/fix-scalability

Automatic merge from submit-queue.

Fix sig-scalability

#1149 updated the README for sig-scalability without modifying sigs.yaml. This fixes that, as well as moves the remaining meeting dates into the custom content bit.

Commit:
Merge pull request #1146 from marun/fix-issue-process-labels

Automatic merge from submit-queue.

Update active milestone issues process to reflect new labels

thockin committed:
2017 Steering Committee election results

Commit:
Merge pull request #1151 from thockin/master

Automatic merge from submit-queue.

2017 Steering Committee election results

jessfraz committed:
Security Process: Add Distributors List Process

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

Commit:
Merge pull request #888 from oracle/for/upstream/master/vault_provider_spec

Automatic merge from submit-queue.

Vaultproviderspec (#1)

* Initial version of Vault KMS Provider
* Renamed image file to be consistent with spec name
* Brought spec inline with EnvelopeTransformer interfaces in PR 49350
* Wrap at 80

Commit:
Initial commit

Commit:
Add section on manually triggering CronJobs

Commit:
Merge pull request #1155 from erhudy/feature/cronjob-manual-trigger

Automatic merge from submit-queue.

Add section on manually triggering CronJobs

As requested by @soltysh, in relation to https://github.com/kubernetes/kubernetes/issues/47538.

Commit:
Merge pull request #1153 from jessfraz/security-process

Automatic merge from submit-queue.

Security Process: Add Distributors List Process

cc @philips @cjcullen @tallclair @liggitt 

First draft based off items in http://oss-security.openwall.org/wiki/mailing-lists/distros

Commit:
Merge pull request #1154 from castrojo/add-election-process

Automatic merge from submit-queue.

Initial commit of the election process

This adds the process we used to run the steering committee election to the community directory so we can refer to it for the election next year.

nikhita committed:
Hardcore spec and status paths

nikhita committed:
validate only status and update Scale type
